### PR TITLE
BotW MQ Rework (+Any New Tricks and Preset Updates)

### PR DESCRIPTION
**Pull request by Joshua8600**

### CODE REVIEW

File: packages/core/lib/combo/settings/tricks.ts [hunk 1]
```diff
@@ -324,6 +324,11 @@ export const TRICKS: Tricks = {
     name: "Enter Zelda's Courtyard using Age Swap in Ganon's Castle",
     tooltip: "Exiting Ganon's Castle as Child puts you in front of the crawlspace to Zelda. This will allow Age Swap to be logical for doing this.",
   },
+  OOT_BOTW_MQ_BACKFLIP: {
+    game: 'oot',
+    name: "Backflip Over Gap to Reach BotW MQ's Main Room's Center",
+    tooltip: "Makes reachable the center of the main room without Zelda's Lullaby (that opens gates) or Hover Boots.",
+  },
```
Comment (XenoWars) [inline packages/core/lib/combo/settings/tricks.ts:331]: Is there really a need to say ZL is what opens the gates? Sidehop is also doable for this.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 2]
```diff
@@ -18,39 +18,21 @@
 "Bottom of the Well Main":
   dungeon: BotW
   exits:
-    "Bottom of the Well Wallmaster Main": "soul_wallmaster && (can_play_zelda || (has_explosives_or_hammer && has_hover_boots))"
-    "Bottom of the Well Wallmaster Basement": "soul_wallmaster"
-    "Bottom of the Well Wallmaster Pit": "soul_wallmaster && (can_hit_triggers_distance || can_boomerang || has_explosives || can_hookshot)"
-    "Bottom of the Well": "true"
+    "Bottom of the Well": "is_child"
+    "Bottom of the Well Main Room Center": "event(BOTW_MQ_CENTER_GATES_LOWERED)"
+    "Bottom of the Well Main Room Center Switch Corner": "has_explosives_or_hammer"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:23]: What about climb or hook anywhere? i imagine at least hook would work.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 3]
```diff
@@ -18,39 +18,21 @@
...
+    "Bottom of the Well Basement": "true"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:24]: Is there an opening to the basement I don't know about? I thought all the holes are sealed up.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 4]
```diff
@@ -18,39 +18,21 @@
...
+    "Bottom of the Well Coffin Room": "(event(BOTW_MQ_DRAIN_WATER) || can_dive_small) && small_keys_botw(2)"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:25]: Surely hook anywhere can do this too, possibly not even anywhere and just normal hook.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 5]
```diff
@@ -59,7 +41,108 @@
...
     "MQ Bottom of the Well Wonder Item Main Room Right 3": "can_use_slingshot"
     "MQ Bottom of the Well Wonder Item Main Room Right 4": "can_use_slingshot"
-    "MQ Bottom of the Well Wonder Item Side Room 1": "(can_play_zelda || (has_explosives_or_hammer && has_hover_boots)) && has_lens && can_use_slingshot"
-    "MQ Bottom of the Well Wonder Item Side Room 2": "(can_play_zelda || (has_explosives_or_hammer && has_hover_boots)) && has_lens && can_use_slingshot"
-    "MQ Bottom of the Well Wonder Item Side Room 3": "(can_play_zelda || (has_explosives_or_hammer && has_hover_boots)) && has_lens && can_use_slingshot"
-    "MQ Bottom of the Well Wonder Item Side Room 4": "(can_play_zelda || (has_explosives_or_hammer && has_hover_boots)) && has_lens && can_use_slingshot"
+"Bottom of the Well Main Room Center Switch Corner":
+  dungeon: BotW
+  exits:
+    "Bottom of the Well Main": "has_explosives_or_hammer"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:47]: Can surely hook or climb anywhere out of here.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 6]
```diff
@@ -59,7 +41,108 @@
...
+  dungeon: BotW
+  exits:
+    "Bottom of the Well Wallmaster Main": "soul_wallmaster"
+    "Bottom of the Well Main Room Center Switch Corner": "has_hover_boots || trick(OOT_BOTW_MQ_BACKFLIP) || hookshot_anywhere || climb_anywhere"
+    "Bottom of the Well West Center Room": "event(BOTW_CENTER_WEST_TORCH)"
+    "Bottom of the Well East Center Room": "event(BOTW_CENTER_EAST_TORCH)"
+    "Bottom of the Well Basement Switch Ledge": "event(BOTW_MQ_GATE_FROM_SWITCH)"
+    "Bottom of the Well Basement": "true"
+  events:
+    BOTW_CENTER_EAST_TORCH: "can_hit_triggers_distance || can_hookshot || has_explosives || can_use_mask_blast"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:62]: i wonder if you can climb anywhere and use a melee weapon.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 7]
```diff
@@ -59,7 +41,108 @@
...
+    "MQ Bottom of the Well GS Basement": "gs && can_damage_skull"
+    "MQ Bottom of the Well Heart Basement 1": "true"
+    "MQ Bottom of the Well Heart Basement 2": "true"
+    "MQ Bottom of the Well Heart Basement 3": "true"
+"Bottom of the Well West Center Room":
+  dungeon: BotW
+  exits:
+    "Bottom of the Well Main Room Center": "true"
+  locations:
+    "MQ Bottom of the Well GS West Middle Room": "gs && can_damage"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:87]: Why are this one and the one just above using different macros?

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 8]
```diff
@@ -59,7 +41,108 @@
...
+    "MQ Bottom of the Well Pot Side Room 3": "true"
+    "MQ Bottom of the Well Wonder Item Side Room 1": "can_use_slingshot"
+    "MQ Bottom of the Well Wonder Item Side Room 2": "can_use_slingshot"
+    "MQ Bottom of the Well Wonder Item Side Room 3": "can_use_slingshot"
+    "MQ Bottom of the Well Wonder Item Side Room 4": "can_use_slingshot"
+"Bottom of the Well Basement Switch Ledge":
+  dungeon: BotW
+  exits:
+    "Bottom of the Well Basement": "true"
+    "Bottom of the Well Main Room Center": "climb_anywhere && hookshot_anywhere"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:105]: does this actually work with the gate still there? Those gates have collision on both sides.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 9]
```diff
@@ -59,7 +41,108 @@
...
+  events:
+    BOTW_MQ_REDEAD_CHEST: "true"
+"Bottom of the Well Coffin Room":
+  dungeon: BotW
+  exits:
+    "Bottom of the Well Main": "true"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:111]: This should require the keys, and now that i think of it should have another region for the side near the door.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 10]
```diff
@@ -59,7 +41,108 @@
...
+"Bottom of the Well Pit Before Cage":
+  dungeon: BotW
+  exits:
+    "Bottom of the Well Main Side Room Before Pit": "true"
+    "Bottom of the Well Cage": "small_keys_botw(2)"
+    "Bottom of the Well Wallmaster Pit": "soul_wallmaster"
+"Bottom of the Well Cage":
+  dungeon: BotW
+  exits:
+    "Bottom of the Well Pit Before Cage": "true"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:146]: This should require the keys.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml [hunk 11]
```diff
@@ -59,7 +41,108 @@
...
+    "MQ Bottom of the Well Pot Lobby Cage 1": "true"
+    "MQ Bottom of the Well Pot Lobby Cage 2": "true"
+    "MQ Bottom of the Well Pot Lobby Cage 3": "true"
+    "MQ Bottom of the Well Lobby Cage Big Fairy": "can_play_sun"
+"Bottom of the Well Basement":
+  dungeon: BotW
+  exits:
+    "Bottom of the Well Main": "true"
+    "Bottom of the Well Basement Switch Ledge": "climb_anywhere || hookshot_anywhere"
+    "Bottom of the Well Wallmaster Basement": "soul_wallmaster"
```
Comment (XenoWars) [inline packages/data/src/world/mq/bottom_of_the_well_mq.yml:74]: Does going straight to the Pits room with climb anywhere work for MQ?

**Review by XenoWars — CHANGES_REQUESTED** [review]

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 3 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml:24]: There is one hole open on the Northeast corner.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 6 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml:62]: I tried before and couldn't manage it.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 2 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: climb does, hook does not. fixed.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 4 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: added hook anywhere.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 5 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: climb does work; fixed.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 9 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: fixed for the keys. can do the ledge region when door rando happens.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 11 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml:79]: no, as you said above the gate is double-sided collision. would not work.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 10 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: fixed

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 8 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: added the gate event to mkae it work.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 2 above)
Comment (XenoWars) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: Hook does. include it.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 5 above)
Comment (XenoWars) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: Hook works. Add it.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 2 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: Added.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 4 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: Added.

File: packages/data/src/world/mq/bottom_of_the_well_mq.yml (hunk 5 above)
Comment (Joshua8600) [reply packages/data/src/world/mq/bottom_of_the_well_mq.yml]: Added.

**Review by XenoWars — APPROVED** [review]

**Pull request merged**